service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -120,6 +124,16 @@ public DownstreamSenderFactory getMessageSenderClient() {
 
     };
 
+    /**
+     * Creates an instance initialized with the given metrics.
+     * 
+     * @param metrics The metrics to use.
+     * @throws NullPointerException if metrics is {@code null}.
+     */
+    public AbstractProtocolAdapterBase(final U metrics) {
```

Comment:
I think this should be protected as the abstract class cannot be instantiated anyway ...